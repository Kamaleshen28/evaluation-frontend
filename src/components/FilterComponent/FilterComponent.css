.search-box {
  padding: 7px;
  border-radius: 10px;
  height: 20px;
  width: 220px;
}

@media only screen and (max-width: 540px) {
  .search-box {
    width: 90px;
  }
}

.search-button {
  padding: 3px;
  border-radius: 5px;
  height: 25px;
  cursor: pointer;
}

.body-header {
  display: flex;
  flex-direction: column;
  gap: 2px;
  padding: 10px;
  width: 100%;
}

@media only screen and (max-width: 400px) {
  .body-header {
    flex-direction: column;
  }
}

.back-button {
  cursor: pointer;
}

.searchbar-lens {
  display: flex;
  align-items: center;
}

.filter-text-with-icon {
  display: flex;
  align-items: center;
}

.filter-icon {
  margin-right: 10px;
  font-size: 27px;
}

.filter-text {
  font-weight: bolder;
  margin-right: 10px;
}

.body-header-top-section {
  display: flex;
  justify-content: space-between;
  margin-bottom: 15px;
}

.radio-button-label {
  font-weight: bolder;
}

.radio-button-container {
  /* display: flex;
  justify-content: space-between; */
}

.all-bookmark-label {
  display: flex;
  justify-content: space-between;
}

.registered-seats-label {
  display: flex;
  justify-content: space-between;
}