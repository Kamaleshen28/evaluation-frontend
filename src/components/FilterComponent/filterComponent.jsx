import React from 'react';
import './FilterComponent.css';
import proptypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';


export default function FilterComponent({ searchInput, handleSearchBoxChange }) {


  const [formData, setFormData] = React.useState(
    {
      filter: '',
    });

  console.log(formData);

  function handleChange(event) {
    const { name, value, type, checked } = event.target;
    setFormData(prevFormData => {
      return {
        ...prevFormData,
        [name]: type === 'checkbox' ? checked : value
      };
    });
  }

  return (
    <div className='body-header'>


      <form >



        {/* <fieldset> */}
        <legend>FILTER</legend>
        <input
          type="radio"
          id="ALL"
          name="filter"
          value="ALL"
          checked={formData.employment === 'ALL'}
          onChange={handleChange}
        />
        <label htmlFor="ALL">ALL</label>
        <br />

        <input
          type="radio"
          id="BOOKMARKED"
          name="filter"
          value="BOOKMARKED"
          checked={formData.employment === 'BOOKMARKED'}
          onChange={handleChange}
        />
        <label htmlFor="BOOKMARKED">BOOKMARKED</label>
        <br />

        <input
          type="radio"
          id="REGISTERED"
          name="filter"
          value="REGISTERED"
          checked={formData.employment === 'REGISTERED'}
          onChange={handleChange}
        />
        <label htmlFor="REGISTERED">REGISTERED</label>
        <br />

        <input
          type="radio"
          id="SEATS AVAILABLE"
          name="filter"
          value="SEATS AVAILABLE"
          checked={formData.employment === 'SEATS AVAILABLE'}
          onChange={handleChange}
        />
        <label htmlFor="SEATS AVAILABLE">SEATS AVAILABLE</label>
        <br />
        {/* </fieldset> */}
        <br />


      </form>

      <input
        className='search-box'
        placeholder='EVENT NAME'
        value={searchInput}
        name='input'
        onChange={handleSearchBoxChange}
      ></input>
      <FontAwesomeIcon icon={faMagnifyingGlass} className='search-button' />





    </div>
  );
}

FilterComponent.propTypes = {
  searchInput: proptypes.string,
  handleSearchBoxChange: proptypes.func
};