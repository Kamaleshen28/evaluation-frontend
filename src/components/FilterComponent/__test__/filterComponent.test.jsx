import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import FilterComponent from '../filterComponent';

jest.mock('react-router-dom');



describe('card', () => {
  it('should render filterComponent correctly', async () => {
    const { asFragment } = render(<FilterComponent />);
    await waitFor(() => {
      expect(screen.getAllByText('ALL')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});



