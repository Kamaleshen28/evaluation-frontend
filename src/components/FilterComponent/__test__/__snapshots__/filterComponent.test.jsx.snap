// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render filterComponent correctly 1`] = `
<DocumentFragment>
  <div
    class="body-header"
  >
    <form>
      <legend>
        FILTER
      </legend>
      <input
        id="ALL"
        name="filter"
        type="radio"
        value="ALL"
      />
      <label
        for="ALL"
      >
        ALL
      </label>
      <br />
      <input
        id="BOOKMARKED"
        name="filter"
        type="radio"
        value="BOOKMARKED"
      />
      <label
        for="BOOKMARKED"
      >
        BOOKMARKED
      </label>
      <br />
      <input
        id="REGISTERED"
        name="filter"
        type="radio"
        value="REGISTERED"
      />
      <label
        for="REGISTERED"
      >
        REGISTERED
      </label>
      <br />
      <input
        id="SEATS AVAILABLE"
        name="filter"
        type="radio"
        value="SEATS AVAILABLE"
      />
      <label
        for="SEATS AVAILABLE"
      >
        SEATS AVAILABLE
      </label>
      <br />
      <br />
    </form>
    <div
      class="searchbar-lens"
    >
      <input
        class="search-box"
        name="input"
        placeholder="EVENT NAME"
        value=""
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass search-button"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
